Docs mysql connector modularization

alex-hat-de-rojo opened the pull request:
@gunnarmorling - hi, am submitting the mysql connector modularization effort here. I have altered the structure of how the mysql page is served in that now it is broken down more neatly into 4 sections. I did need to alter my forked site playbook file to serve but I believe that should not impact how it is delivered on the proper site. I know it is a big pull request for docs and I have sent an email as well.

sixsixsix516 commented:
hi , i have a question . 
I can't find the button to submit a question to this warehouse
I don't know how to use GitHub
How do you submit your questions
pealse, thanks !!!

gunnarmorling commented:
Hey @alex-hat-de-rojo, could we do a call right about this one tomorrow? E.g. 11.00 a.m. CET? Just to get a better understanding of the changes. Thx!

alex-hat-de-rojo commented:
@Gunnar Morling <gmorling@redhat.com> - of course. Could we push it
sometime from 1400 CET onward? I'm Stateside for one more week so just to
be sure I'm awake enough. :)

Am Mo., 18. Nov. 2019 um 06:56 Uhr schrieb Gunnar Morling <
notifications@github.com>:

> Hey @alex-hat-de-rojo <https://github.com/alex-hat-de-rojo>, could we do
> a call right about this one tomorrow? E.g. 11.00 a.m. CET? Just to get a
> better understanding of the changes. Thx!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/debezium/debezium/pull/1138?email_source=notifications&email_token=AL6YIZBCVZPGXF5IONEH64LQUKGHFA5CNFSM4JNCGEKKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEKK5RY#issuecomment-555003591>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL6YIZCV4MF4G3KMF6GV6KTQUKGHFANCNFSM4JNCGEKA>
> .
>


-- 

Alex mansell

Technical Writer, Customer Content Services

Red Hat
<https://www.redhat.com>

cmansell@redhat.com    M: 353-87-141-7699 <+353-87-141-7699>

gunnarmorling commented:
Ha, didn't know you'd be in the US :) I'll set something up in BlueJeans in my afternoon then.

gunnarmorling commented on documentation/modules/ROOT/pages/_attributes.adoc:
```diff
@@ -11,3 +11,9 @@
+:assemblies: ../assemblies
+:modules: ../../modules
+
+:mysql-connector-plugin-download: https://repo1.maven.org/maven2/io/debezium/debezium-connector-mysql/1.0.0.Beta2/debezium-connector-mysql-1.0.0.Beta2-plugin.tar.gz
```
The version should refer to the "debezium-version" variable instead of hard-coding it.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -324,7 +324,7 @@ Each message written to the schema change topic will have a message key that con
   {
     "schema": {
       "type": "struct",
-      "name": "io.debezium.connector.mysql.SchemaChangeKey",
```
I don't think we'd ever replace these type names with something product-specific. It's the same as for Strimzi, where "strimzi" e.g. shows in CRD version names.

Naros replied:
Wouldn't this substitution change yield `io.Debezium.connector.mysql.SchemaChangeKey` rather than the expected name of `io.debezium.connector.mysql.SchemaChangeKey`?  I noticed this in a few places such as class names and URLs for example where we are substituting `debezium` with `{prodname}` which is defined as `Debezium`.

gunnarmorling replied:
Yes, exactly. This one shouldn't be substituted.

gunnarmorling commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -891,7 +891,7 @@ The following table describes how the connector maps each of the MySQL data type
 
 |`BIT( > 1)`
 |`BYTES`
-|`io.debezium.data.Bits`
```
Same comment as above, this replacement went a bit too far.

gunnarmorling commented on documentation/modules/ROOT/pages/modules/cdc-mysql-connector/r_mysql-supported-topologies.adoc:
```diff
@@ -0,0 +1,39 @@
+
+Multi-master::
+====
+A link:https://dev.mysql.com/doc/refman/5.7/en/mysql-cluster-replication-multi-master.html[multi-master MySQL topology] uses one or more MySQL slaves that each replicate from multiple masters. This is a powerful way to aggregate the replication of multiple MySQL clusters, and requires using GTIDs.
```
Would be good to replace the MySQL version with a variable globally and refer to that. It also should be 8.0 by now (or "stable", if they have some alias like that).

gunnarmorling commented:
Alex, one general question: this PR adds 2K lines net, whereas I'd have expected the same number of lines to be removed from the original monolithic file, replaced with includes to the newly extracted fragments. What am I missing?

alex-hat-de-rojo commented:
@gunnarmorling - I've incorporated the following based on our chat:

- the `{prodname}` attribute has been removed from code snippets
- mysql version is now an attribute (currently reflecting 8.0)
- removed the original mysql connector .adoc file

**Note**: I haven't yet set the TOC macro as there's some small issues I need to work out there, but as soon as I have it, I'll let you know.

gunnarmorling commented:
@alex-hat-de-rojo, could you rebase this so to resolve the merge conflict; also to be sure: will the existing .../connectors/mysql URL still work? It should probably just be some kind of basic landing page for that connector now.

alex-hat-de-rojo commented:
@gunnarmorling - it looks like I don't have the permissions to do that?

gunnarmorling commented:
> it looks like I don't have the permissions to do that?

I'm not sure I'm following, do you get any kind of error message? Which one? In a nutshell, here's what should work (all commands to be run in your local working copy):

```
git checkout master
git pull ustream master
git checkout DOCS-mysqlConnector-modularization
git rebase master # resolve conflicts manually
git push -f origin DOCS-mysqlConnector-modularization # force-push the branch of this PR
```

gunnarmorling commented:
@alex-hat-de-rojo Hey there, how is it going? Does the rebase work out for you?

gunnarmorling commented:
Closing this in favour of #1152. Thanks, @alex-hat-de-rojo!
